src/toil/provisioners/clusterScaler.py

```diff
@@ -235,6 +235,30 @@ def __init__(self, provisioner, jobBatcher, config):
         else:
             self.scaler = None
 
+    def check(self):
+        """
+        Attempt to join any existing scaler threads with a short timeout. This insures an exception
+        raised in the threads is propagated in a timely fashion. If there are exceptions in both threads,
+        one exception is logged before the other is raised to insure that both tracebacks are visible
+        """
+        exceptions = []
+        if self.preemptableScaler:
+            try:
+                self.preemptableScaler.join(1)
+            except Exception as e:
+                exceptions.append(e)
```

Comment:
I think to get an accurate backtrace, you need to log the exception right away with log.error(…,excinfo=True). That is because the backtrace is not part of the exception, but stored in a global variable called [sys.exc_info](https://docs.python.org/2/library/sys.html#sys.exc_info) (one of Python's design flaws).